docker/runtime/backup-init.sh

```diff
@@ -0,0 +1,138 @@
+
+	if [ "${TIME_UNIT}" = "h" ]; then
+		if [ ! "${TIME_INTERVAL}" -le 23 ]; then
+			echo "Parse error: Time unit 'h' (hour) largest value is 23" 1>&2
+			exit 1
+		fi
+
+		CRON_EXPR_MINUTES=0
+		CRON_EXPR_HOURS="*/${TIME_INTERVAL}"
+	elif [ "${TIME_UNIT}" = "d" ]; then
+		if [ ! "${TIME_INTERVAL}" -le 30 ]; then
+			echo "Parse error: Time unit 'd' (day) largest value is 30" 1>&2
+			exit 1
+		fi
+
+		CRON_EXPR_MINUTES=0
+		CRON_EXPR_HOURS=0
+		CRON_EXPR_DAYS="*/${TIME_INTERVAL}"
+	elif [ "${TIME_UNIT}" = "M" ]; then
+		if [ ! "${TIME_INTERVAL}" -le 12 ]; then
+			echo "Parse error: Time unit 'M' (month) largest value is 12" 1>&2
+			exit 1
+		fi
+
+		CRON_EXPR_MINUTES=0
+		CRON_EXPR_HOURS=0
+		CRON_EXPR_DAYS="1"
+		CRON_EXPR_MONTHS="*/${TIME_INTERVAL}"
+	else
+		echo "Parse error: BACKUP_INTERVAL expression is invalid" 1>&2
+		exit 1
+	fi
+
+	echo "${CRON_EXPR_MINUTES} ${CRON_EXPR_HOURS} ${CRON_EXPR_DAYS} ${CRON_EXPR_MONTHS} *"
+}
+
+# Parse BACKUP_RETENTION environment variable and generate appropriate find command expression.
+# Expected format: nu (n - number, u - unit) (eg. 3d equals 3 days)
+# Supported units: m - minutes, d - days
+parse_generate_retention_expression() {
+	FIND_TIME_EXPR='mtime'
+
+	TIME_INTERVAL=$(echo "${BACKUP_RETENTION}" | sed -e 's/[mhdM]$//')
+	TIME_UNIT=$(echo "${BACKUP_RETENTION}" | sed -e 's/^[0-9]\+//')
+
+	if [ "${TIME_UNIT}" = "m" ]; then
+		if [ "${TIME_INTERVAL}" -le 59 ]; then
+			echo "Warning: Minimal retention is 60m. Value set to 60m" 1>&2
+			TIME_INTERVAL=60
+		fi
+
+		FIND_TIME_EXPR="mmin"
+	elif [ "${TIME_UNIT}" = "h" ]; then
+		echo "Error: Unsupported expression - Try: eg. 120m for 2 hours." 1>&2
+		exit 1
+	elif [ "${TIME_UNIT}" = "d" ]; then
+		FIND_TIME_EXPR="mtime"
+	elif [ "${TIME_UNIT}" = "M" ]; then
+		echo "Error: Unsupported expression - Try: eg. 60d for 2 months." 1>&2
+		exit 1
+	else
+		echo "Parse error: BACKUP_RETENTION expression is invalid" 1>&2
+		exit 1
+	fi
+
+	echo "${FIND_TIME_EXPR} +${TIME_INTERVAL:-7}"
+}
+
+add_backup_cronjob() {
+	CRONTAB_USER="${1:-git}"
+	CRONTAB_FILE="/etc/crontabs/${CRONTAB_USER}"
+	CRONJOB_EXPRESSION="${2:-}"
+	CRONJOB_EXECUTOR="${3:-}"
+	CRONJOB_EXECUTOR_ARGUMENTS="${4:-}"
+  CRONJOB_TASK="${CRONJOB_EXPRESSION} /bin/sh ${CRONJOB_EXECUTOR} ${CRONJOB_EXECUTOR_ARGUMENTS}"
+
+	if [ -f "${CRONTAB_FILE}" ]; then
+		CRONJOB_EXECUTOR_COUNT=$(grep -c "${CRONJOB_EXECUTOR}" "${CRONTAB_FILE}" || exit 0)
+		if [ "${CRONJOB_EXECUTOR_COUNT}" != "0" ]; then
+			echo "Cron job already exists for ${CRONJOB_EXECUTOR}. Updating existing." 1>&2
+			CRONJOB_TASK=$(echo "{CRONJOB_TASK}" | sed 's/\//\\\//g' )
+			CRONJOB_EXECUTOR=$(echo "{CRONJOB_EXECUTOR}" | sed 's/\//\\\//g' )
+			sed -i "/${CRONJOB_EXECUTOR}/c\\${CRONJOB_TASK}" "${CRONTAB_FILE}"
+			return 0
+		fi
+	fi
+
+	# Finally append new line with cron task expression
+	echo "${CRONJOB_TASK}" >>"${CRONTAB_FILE}"
+}
+
+CRONTAB_USER=$(awk -v val="${PUID}" -F ":" '$3==val{print $1}' /etc/passwd)
+
+set +e
```

Comment:
Can you add a comment to explain the reason to have this line?

Reply:
Absolutely. In a bit.